src/libs/kata-types/src/annotations/mod.rs

```diff
@@ -12,3 +22,884 @@ pub mod crio;
+
+// Runtime related annotations
+/// Prefix for Runtime configurations.
+pub const KATA_ANNO_CFG_RUNTIME_PREFIX: &str = "io.katacontainers.config.runtime.";
+/// A sandbox annotation that determines if seccomp should be applied inside guest.
+pub const KATA_ANNO_CFG_DISABLE_GUEST_SECCOMP: &str =
+    "io.katacontainers.config.runtime.disable_guest_seccomp";
+/// A sandbox annotation that determines if pprof enabled.
+pub const KATA_ANNO_CFG_ENABLE_PPROF: &str = "io.katacontainers.config.runtime.enable_pprof";
+/// A sandbox annotation that determines if experimental features enabled.
+pub const KATA_ANNO_CFG_EXPERIMENTAL: &str = "io.katacontainers.config.runtime.experimental";
+/// A sandbox annotaion that determines how the VM should be connected to the the container network
+/// interface.
+pub const KATA_ANNO_CFG_INTER_NETWORK_MODEL: &str =
+    "io.katacontainers.config.runtime.internetworking_model";
+/// SandboxCgroupOnly is a sandbox annotation that determines if kata processes are managed only in sandbox cgroup.
+pub const KATA_ANNO_CFG_SANDBOX_CGROUP_ONLY: &str =
+    "io.katacontainers.config.runtime.sandbox_cgroup_only";
+/// A sandbox annotation that determines if create a netns for hypervisor process.
+pub const KATA_ANNO_CFG_DISABLE_NEW_NETNS: &str =
+    "io.katacontainers.config.runtime.disable_new_netns";
+/// A sandbox annotation to specify how attached VFIO devices should be treated.
+pub const KATA_ANNO_CFG_VFIO_MODE: &str = "io.katacontainers.config.runtime.vfio_mode";
+
+/// A helper structure to query configuration information by check annotations.
+#[derive(Debug, Default, Deserialize)]
+pub struct Annotation {
+    annotations: HashMap<String, String>,
+}
+
+impl From<HashMap<String, String>> for Annotation {
+    fn from(annotations: HashMap<String, String>) -> Self {
+        Annotation { annotations }
+    }
+}
+
+impl Annotation {
+    /// Create a new instance of [`Annotation`].
+    pub fn new(annotations: HashMap<String, String>) -> Annotation {
+        Annotation { annotations }
+    }
+
+    /// Deserialize an object from a json string.
+    pub fn deserialize<T>(path: &str) -> Result<T>
+    where
+        for<'a> T: Deserialize<'a>,
+    {
+        let f = BufReader::new(File::open(path)?);
+        Ok(serde_json::from_reader(f)?)
+    }
+
+    /// Get an immutable reference to the annotation hashmap.
+    pub fn get_annotations(&self) -> &HashMap<String, String> {
+        &self.annotations
+    }
+
+    /// Get a mutable reference to the annotation hashmap.
+    pub fn get_annotations_mut(&mut self) -> &mut HashMap<String, String> {
+        &mut self.annotations
+    }
+
+    /// Get the value of annotation with `key` as string.
+    pub fn get(&self, key: &str) -> Option<String> {
+        self.annotations.get(key).map(|v| String::from(v.trim()))
+    }
+
+    /// Get the value of annotation with `key` as bool.
+    pub fn get_bool(&self, key: &str) -> Result<Option<bool>> {
```

Comment:
get_bool/get_u32/get_i32/get_u64 can use generic type like this:

```rust
    pub fn get_value<T>(&self, key: &str) -> Result<Option<T>, <T as std::str::FromStr>::Err>
    where
        T: std::str::FromStr,
    {
        if let Some(value) = self.get(key) {
            return value.parse::<T>().map(Some);
        }
        Ok(None)
    }

```

And use as:

```rust
self.get_value::<i32>(KATA_ANNO_CONTAINER_RES_SWAPPINESS)
```

Reply:
fixed this in next version